src/MicrosoftTeams.ts

```diff
@@ -572,35 +589,43 @@ namespace microsoftTeams {
         handlers["authentication.authenticate.failure"] = handleFailure;
 
         /**
-         * Initiates an authentication request, which opens a new window with the specified settings.
+         * Registers the authentication parameters
          * @param authenticateParameters A set of values that configure the authentication pop-up.
+	     */
+        export function registerAuthenticationParameters(authenticateParameters: AuthenticateParameters): void {
```

Comment:
The authentication parameters(successCallback method etc) are registered via the authenticate() call but in Win32 Outlook when we redirect the page back to connector's url, the iframe page also reloads. 
This means the 3rd party needs to register the authenticate parameters again so that the connector's page can call the callback methods. Since authenticate() call only happens when the user clicks on Authenticate button, we need an API to register the authentication params automatically on page load. 
So developers now need to call registerAuthenticationParameters() in addition to initialize() on the configuration page to ensure the successCallback/failureCallback methods are registered in the SDK on page reload. This also means they no longer need to pass these parameters in the authenticate() call.

Reply:
Why not just make this an optional parameter to the initialize call then? Also, I think a better name for this would be something like registerAuthenticationHandlers. What happens if the parent Connectors page invokes the callbacks before this code has run? Will the app just fail to get notified?

Reply:
If we make it optional parameter in the initialize call, we'd have to expose the AuthenticateParameters interface outside the Authentication namespace. Although that would ensure that callbacks would be invoked only after the handlers are registered but it doesn't look like an ideal design.
Yes, the successCallback/failureCallback methods would not be called if the handlers are invoked by connector's page before this API is called. We can document that this API needs to be called even before initialize() on this page to prevent this.
I'm fine with either approach.